flang/runtime/edit-input.cpp

```diff
@@ -244,7 +244,15 @@ bool EditIntegerInput(
     value = -value;
   }
   if (any || !io.GetConnectionState().IsAtEOF()) {
-    std::memcpy(n, &value, kind); // a blank field means zero
+    // The value is stored in the lower order bits on big endian platform.
+    // When memcpy, shift the value to the higher order bit.
+    auto shft{static_cast<int>(sizeof(value.low())) - kind};
+    if (!isHostLittleEndian && shft >= 0) {
```

Comment:
Can be `shft > 0`.

Reply:
We still need `shft==0` case. It is when `kind==8`, the value is still stored in `low_`.